distributed/tests/test_spill.py

```diff
@@ -218,4 +218,3 @@ def __sizeof__(self):
     # Because before trying to write to disk, we try to serialize to get the size to see if it's bigger
     # than max spill. Hence we are having a  WARNING - Failed to pickle
     buf["a"] = a
```

Comment:
@crusaderky I initially try to go over this, but the main issue I have is that we try to serialize before the check for Max Spill. Therefore for case 1-a I trigger the serialize error and it returns, losing the keys.  

Key bigger than spill threshold, tries to write to disk, and the first thing we do is try to serialize, so I hit the error and it returns. 

https://github.com/ncclementi/distributed/blob/52fd06c5fb2436926d58de0c7ad23235a46ff396/distributed/spill.py#L105-L109

My understanding was that in this case we wanted to keep the keys in fast. Unless this is incorrect.

From what I see having a bad key that exceeds spill threshold, will trigger the serialization error first, both in the case max-spill is passed or not.

Reply:
The only case where you are going to be using a class Bad with a ``__getstate__`` that raises is 1c/2c.

To test 1a/2a, you can just use a vanilla ``bytes`` object.
To test 1b/2b, you can use a vanilla bytes object AND write to a read-only ``tmpdir``.